### DISCUSSION TITLE
Version updates

### DISCUSSION THREAD

**Discussion by TonyGravagno** [araera111/mysql-to-zod#152] [General]
Since we are both making changes to the package, and frequently, it might help if there is a build bump in the version ID with each new push to master. What do you think about using the date with a sequential index?

- 2.2.08291 < index 1
- 2.2.08301 < new day, restart index
- 2.2.08302 < same day, second update
- 2.2.08303 < third update
- 2.2.08311 < new day, restart index

I actually don't like that idea too much because it requires manual effort ... unless this is done automatically on push to main/master?

Just using a simple index should be adequate as well: 2.3.1, 2.3.2 ...

At this point you might also want to consider what change is worthy of a bump to the minor version: 2.4.0, 2.5.0 ...

Another approach to this is to have a 'dev' branch that gets all of the build updates that you approve. We can continue to pull and PR to that branch. The main branch only gets updated when a number of features have been added.

Also consider : When dev is stable and enough versions have been posted there, branch from dev to 'rc0801' (release candidate month 08, first 01 candidate. Documentation can be finalized, final testing can be done, and that is the version that will get pushed to main, not 'dev' ... along with a bump to the minor version.

Thoughts?

**Comment by araera111** [comment]
Thank you for your thoughtful suggestions regarding versioning and branching strategies.

I have been following semantic versioning initially, but it seems I have recently overlooked updating the version numbers. Implementing a develop branch sounds like a promising approach for managing collaborative work. I will proceed with creating the develop branch.

To clarify, fixes will be applied directly to main, while feature developments will be branched from develop and then merged into main.

Thank you for your input. It’s greatly appreciated as we work to streamline our process.
